src/main/java/org/kuali/kfs/pdp/businessobject/service/ACHPayeeSearchService.java

```diff
@@ -0,0 +1,347 @@
+
+        if (payeeTypeEntered
+            && PdpConstants.PayeeIdTypeCodes.VENDOR_ID.equals(payeeTypeCode)
+            && isEmployeeInfoEntered) {
+            final String messageKey = PdpKeyConstants.ERROR_PAYEE_LOOKUP_VENDOR_EMPLOYEE_CONFUSION;
+
+            final String employeeIdLabel = getAttributeLabel(KIMPropertyConstants.Person.EMPLOYEE_ID);
+            final String entityIdLabel = getAttributeLabel(KIMPropertyConstants.Person.ENTITY_ID);
+            final String payeeTypeLabel = getAttributeLabel(KFSPropertyConstants.PAYEE_TYPE_CODE);
+
+            GlobalVariables.getMessageMap().putError(KFSPropertyConstants.PAYEE_TYPE_CODE, messageKey, payeeTypeLabel,
+                    payeeTypeCode, employeeIdLabel, entityIdLabel);
+        } else if (payeeTypeEntered
+                   && (PdpConstants.PayeeIdTypeCodes.EMPLOYEE.equals(payeeTypeCode)
+                       || PdpConstants.PayeeIdTypeCodes.ENTITY.equals(payeeTypeCode)) && isVendorInfoEntered) {
+            final String messageKey = PdpKeyConstants.ERROR_PAYEE_LOOKUP_VENDOR_EMPLOYEE_CONFUSION;
+
+            final String vendorNameLabel = getAttributeLabel(KFSPropertyConstants.VENDOR_NAME);
+            final String vendorNumberLabel = getAttributeLabel(KFSPropertyConstants.VENDOR_NUMBER);
+            final String payeeTypeLabel = getAttributeLabel(KFSPropertyConstants.PAYEE_TYPE_CODE);
+
+            GlobalVariables.getMessageMap().putError(KFSPropertyConstants.PAYEE_TYPE_CODE, messageKey, payeeTypeLabel,
+                    payeeTypeCode, vendorNameLabel, vendorNumberLabel);
+        }
+
+        if (GlobalVariables.getMessageMap().hasErrors()) {
+            throw new ValidationException("errors in search criteria");
+        }
+    }
+
+    // Scope loosened for CSU
+    protected String getAttributeLabel(final String attributeName) {
+        return dataDictionaryService.getAttributeLabel(ACHPayee.class, attributeName);
+    }
+
+    // Scope  loosened for CSU
+    protected List<DisbursementPayee> getVendorsAsPayees(final MultiValueMap<String, String> fieldValues) {
+        final List<DisbursementPayee> payeeList = new ArrayList<>();
+
+        final Map<String, String> fieldsForLookup = getVendorFieldValues(fieldValues.toSingleValueMap());
+        vendorLookupable.setBusinessObjectClass(VendorDetail.class);
+        vendorLookupable.validateSearchParameters(fieldsForLookup);
+
+        final List<? extends BusinessObject> vendorList = vendorLookupable.getSearchResults(fieldsForLookup);
+        for (final BusinessObject vendor : vendorList) {
+            final VendorDetail vendorDetail = (VendorDetail) vendor;
+            final DisbursementPayee payee = getPayeeFromVendor(vendorDetail);
+            payeeList.add(payee);
+        }
+
+        return payeeList;
+    }
+
+    //CU customization change from private to protected
+    protected ACHPayee getPayeeFromVendor(final VendorDetail vendorDetail) {
```

Comment:
Do you still need to override the visibility of this method? Can it be reverted back to the visibility from base code?

Reply:
updated